### ISSUE TITLE
Move leeches to deck feature not working

### ISSUE THREAD

**Issue by eternalstudent7** [iamjustkoi/LeechToolkit#28]
**Bug/Issue Description**
Move To deck for new leeches not working

**To Reproduce**
Check "move to deck" within options and give it a designated deck
After seeing "card is a leech" dialog, check to see whether it has been moved to the 

I'm seeing a few other bugs. I can't get the custom tag to work and am not clear how it is supposed to function. I can potentially help with some troubleshooting and testing if someone wants to work on these bugs but I don't code myself. The unleech and lapse decrease features seem to be working ok.

**Comment by AdriGrana** [comment]
Can confirm it isn't working for me either. Hope this will be fixed one day.

**Comment by ajcr07** [comment]
Yeah "actions for leeches" does not work in ver 24.06.3 on my end either (also tested 2.1.66 seems to be the same). That includes moving to Deck, flag or tag etc. 

Curiously I discovered that if you perform the action from the browser (Right click on a card, and choose Toolkit Actions: Leech or Un-Leech) it does work correctly, seems like the action doesn't get triggered from a normal review when the Leech threshold is reached like it should.

For now I just manually perform the "Leech Action" from the browser on leeches and that sends the cards to the specified deck, and applies the rest of the options that I've chosen. I did discover that using settings only specific for a Deck using the legacy options window leads to a _"maximum recursion depth exceeded while calling a Python object"_ error so it's best to stick with the global options.

Oh also the "Forget" action doesn't work, it doesn't make the card new but it does reset the lapses.

**Comment by AdriGrana** [comment]
@ajcr07 

Kinda off-topic but since you use the addon too I have to ask this, does the addon also mess with when the new cards are shown? For me when I have the addon enable the new cards all show up almost at the very end of my reviews (l have like 200 reviews and all 10 new cards come up in the last 30 cards). It's definitely the addon that causes it, now I have it disabled and it doesn't happen again. (it's a deal breaker for me such a bug). Ill maybe open a new issue if I am not the only one who experiences this, but I feel like the dev does not have time to really work on it, so it's probably just one more thing on the list he will never find time to fix.

**Comment by ajcr07** [comment]
@AdriGrana I tried to recreate your bug and I was unable to do it in ver 2.1.66 but in ver 24.06.3 I was able to get that behavior with 1 deck but not with others. I don't know if it's because ver 24.06.3 is using FSRS or is just something broken in later versions, but either way it's not consistent, and honestly neither are the other bugs since this addon apparently works now in one of my decks but not the others, it just happened randomly, and it left me scratching my head.

I can live with these bugs but it does feel like they're beginning to pile up sadly, hopefully newer versions of anki don't end up completely breaking it up entirely since I like it a lot even in this state.

**Comment by AdriGrana** [comment]
I use version 23.12.1 in Anki, so maybe that explains these bugs. Well I stopped using the addon for now anyways as it also messes with the scheduling of new cards for some weird reason (maybe Ill open a new issue but given that the dev doesn't have time to work on it Ill maybe just leave it be), and honestly I don't want a black box that messes with card scheduling, too risky for me.
